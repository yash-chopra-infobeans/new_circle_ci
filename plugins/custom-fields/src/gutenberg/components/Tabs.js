import PropTypes from 'prop-types';

const { TabPanel } = wp.components;

const Tabs = ({ tabs, children }) => {
  if (!tabs || tabs?.length === 0) {
    return children(false);
  }

  return (
    <TabPanel className="cf-tabs" tabs={tabs}>
      {tab => children(tab)}
    </TabPanel>
  );
};

Tabs.propTypes = {
  tabs: PropTypes.array.isRequired,
  children: PropTypes.func.isRequired,
};

export default Tabs;
