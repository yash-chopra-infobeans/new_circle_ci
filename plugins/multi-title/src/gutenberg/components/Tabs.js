import TextareaAutosize from 'react-autosize-textarea';
import { isEmpty } from 'lodash-es';
import TabComp from '../containers/TabContainer';

const { __ } = wp.i18n;
const { TabPanel, Slot, Fill, PanelBody } = wp.components;
const { applyFilters } = wp.hooks;
const { forwardRef } = wp.element;

export const Tabs = forwardRef((props, ref) => {
  const {
    errorMessages = {},
    activeTab,
    selectTab,
    tabs,
    onError,
    handleSubtitleChange,
    subtitles,
  } = props;
  // below filter allows developers to enable/disable standfirst
  const standfirst = applyFilters('multi_title_standfirst_enabled', false);
  const errors = Object.keys(errorMessages).reduce((obj, key) => {
    const newObj = obj;

    if (!errorMessages[key].preventPublish) {
      return newObj;
    }

    newObj[key] = errorMessages[key];

    return newObj;
  }, {});

  return (
    <div>
      {/*  if errors display them within the publishing panel
        so users know why they cannot publish */}
      {!isEmpty(errors) && (
        <Fill name="PluginPrePublishPanel">
          <PanelBody title={__('Errors', 'multi-title')}>
            {Object.keys(errorMessages).map(key => (
              <div className="publishErrors-section">
                <p>
                  <strong>{errorMessages[key].title}</strong>
                </p>
                <ul>
                  <li>{errorMessages[key].message}</li>
                </ul>
              </div>
            ))}
          </PanelBody>
        </Fill>
      )}
      <TabPanel
        className="title-tab-panel"
        activeClass="active-tab"
        tabs={tabs}
        onSelect={selectTab}
      >
        {tab => <TabComp {...props} activeTab={activeTab} tab={tab} tabs={tabs} ref={ref} />}
      </TabPanel>
      <Slot name="multi-title-below-tabs" />
      {standfirst && (
        <TextareaAutosize
          rows={1}
          placeHolder="Standfirst"
          className="title-input title-input-sm title-input-standfirst"
          onChange={e => handleSubtitleChange('standfirst')(e.target.value)}
          onError={onError}
          value={subtitles?.standfirst}
        />
      )}
    </div>
  );
});

export default Tabs;
