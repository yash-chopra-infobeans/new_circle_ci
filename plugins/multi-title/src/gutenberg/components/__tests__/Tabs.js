import { shallow } from 'enzyme';
import { Tabs } from '../Tabs';

describe('<Tabs />', () => {
  it('should render without errors', () => {
    const componentProps = {
      attributes: {
        titles: {
          headline: {
            value: 'test headline',
            prefix: 'test prefix',
          },
        },
      },
      errorMessages: {},
      activeTab: 'headline',
      selectTab: jest.fn(),
      tabs: [],
      onError: jest.fn(),
      handleSubtitleChange: jest.fn(),
    };
    const component = shallow(<Tabs {...componentProps} />);
    const slot = component.find('Slot');

    expect(component).toMatchSnapshot();
    expect(slot.prop('name')).toEqual('multi-title-below-tabs');
  });

  it('should render with errors', () => {
    const componentProps = {
      attributes: {
        titles: {
          headline: {
            value: 'test headline',
            prefix: 'test prefix',
          },
        },
      },
      errorMessages: {
        seo: {
          message: 'You have exceeded the maximum character limit.',
          preventPublish: true,
          title: 'SEO',
        },
      },
      activeTab: 'headline',
      selectTab: jest.fn(),
      tabs: [],
      onError: jest.fn(),
      handleSubtitleChange: jest.fn(),
    };
    const component = shallow(<Tabs {...componentProps} />);
    const errorTitle = component.find('.publishErrors-section p strong');
    const errorText = component.find('.publishErrors-section ul li');

    expect(component).toMatchSnapshot();
    expect(errorTitle.text()).toEqual('SEO');
    expect(errorText.text()).toEqual('You have exceeded the maximum character limit.');
  });
});
