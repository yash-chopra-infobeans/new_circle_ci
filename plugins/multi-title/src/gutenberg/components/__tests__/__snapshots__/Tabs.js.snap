// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render with errors 1`] = `
<div>
  <Fill
    name="PluginPrePublishPanel"
  >
    <ForwardRef(PanelBody)
      title="Errors"
    >
      <div
        className="publishErrors-section"
      >
        <p>
          <strong>
            SEO
          </strong>
        </p>
        <ul>
          <li>
            You have exceeded the maximum character limit.
          </li>
        </ul>
      </div>
    </ForwardRef(PanelBody)>
  </Fill>
  <WithInstanceId(TabPanel)
    activeClass="active-tab"
    className="title-tab-panel"
    onSelect={[MockFunction]}
    tabs={Array []}
  >
    <Component />
  </WithInstanceId(TabPanel)>
  <Slot
    name="multi-title-below-tabs"
  />
</div>
`;

exports[`<Tabs /> should render without errors 1`] = `
<div>
  <WithInstanceId(TabPanel)
    activeClass="active-tab"
    className="title-tab-panel"
    onSelect={[MockFunction]}
    tabs={Array []}
  >
    <Component />
  </WithInstanceId(TabPanel)>
  <Slot
    name="multi-title-below-tabs"
  />
</div>
`;
